src/main/scala/com/target/data_validator/validator/RowBased.scala

```diff
@@ -1,43 +1,96 @@
 package com.target.data_validator.validator
 
-import com.target.data_validator.{ValidatorCheckEvent, ValidatorError, ValidatorQuickCheckError, VarSubstitution}
-import com.target.data_validator.JsonEncoders.eventEncoder
-import com.target.data_validator.validator.ValidatorBase.{isColumnInDataFrame, I0, L0, L1}
-import io.circe.Json
-import io.circe.syntax._
+import com.target.data_validator.{ValidatorCheckEvent, ValidatorCounter, ValidatorError, ValidatorQuickCheckError}
+import com.target.data_validator.VarSubstitution
+import com.target.data_validator.validator.ValidatorBase.{isColumnInDataFrame, L0, L1}
 import org.apache.spark.sql.{DataFrame, Row}
-import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
 import org.apache.spark.sql.catalyst.expressions._
-import org.apache.spark.sql.types.{NumericType, StructType}
+import org.apache.spark.sql.types.StructType
+
+import scala.util.matching.Regex
 
 abstract class RowBased extends CheapCheck {
 
   val column: String
+  val threshold: Option[String]
 
-  def configCheck(df: DataFrame): Boolean = configCheckColumn(df)
+  def configCheck(df: DataFrame): Boolean = {
+    configCheckColumn(df)
+    configCheckThreshold
+    failed
+  }
 
   def configCheckColumn(df: DataFrame): Boolean = {
     if (isColumnInDataFrame(df, column)) {
       logger.debug(s"Column: $column found in table.")
       false
     } else {
-      val msg = s"Column: $column not found in table."
+      val msg = s"Column: $column not found in schema."
       logger.error(msg)
       addEvent(ValidatorError(msg))
       failed
     }
   }
 
+  def configCheckThreshold: Boolean = {
+    if (threshold.isDefined) {
+      val ret = threshold.flatMap(RowBased.THRESHOLD_NUMBER_REGEX.findFirstIn).isEmpty
+      if (ret) {
+        val msg = s"Threshold `${threshold.get}` not parsable."
+        logger.error(msg)
+        addEvent(ValidatorError(msg))
+      }
+      ret
+    } else {
+      false
+    }
+  }
+
   def colTest(schema: StructType, dict: VarSubstitution): Expression
 
   def select(schema: StructType, dict: VarSubstitution): Expression = If(colTest(schema, dict), L1, L0)
 
+  /**
+    * Calculates the max acceptable number of errors from threshold and rowCount.
+    * @param rowCount of table.
+    * @return max number of errors we can tolerate.
+    * if threshold < 0, then its a percentage of rowCount.
+    * if threshold ends with '%' then its percentage of rowCount
+    * if threshold is > 1, then its maxErrors.
+    */
+  def calcErrorCountThreshold(rowCount: Long): Long = {
+    threshold.map { t =>
+      val tempThreshold = t.stripSuffix("%").toDouble
+      val ret: Long = if (t.endsWith("%")) {
+        // Has '%', so divide by 100.
+        (tempThreshold * (rowCount / 100)).toLong
```

Comment:
Why integer division here?  What if rowCount is 99 (I know, not likely)?  This will always be zero then.

Reply:
Bug, added test and fixed it.